### PR TITLE
Makes it possible to bind-mount read-only

### PR DESCRIPTION
**Pull request by tobsan**
Bind-mounting read-only into the container now works, by re-mounting
from inside the container.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

**Review by thenor — APPROVED** [review]

File: libsoftwarecontainer/src/container.cpp [hunk 1]
```diff
@@ -601,7 +605,31 @@ ReturnCode Container::bindMountFolderInContainer(const std::string &pathOnHost,
...
+        return remountReadOnlyInContainer(result);
+    } else {
+        return ReturnCode::SUCCESS;
+    }
+}
+
+ReturnCode Container::remountReadOnlyInContainer(const std::string &path)
+{
+    pid_t pid = INVALID_PID;
+    EnvironmentVariables envVars;
```
Comment (breakreturn) [inline libsoftwarecontainer/src/container.cpp]: The env var 'variables' parameter in executeInContainer has a default value so if we don't plan to pass anything here I guess we can remove the env vars from this part of the code.

File: libsoftwarecontainer/src/container.cpp [hunk 2]
```diff
@@ -574,7 +574,11 @@ ReturnCode Container::bindMountFileInContainer(const std::string &pathOnHost,
     }
 
     result = gatewaysDirInContainer() + "/" + pathInContainer;
-    return ReturnCode::SUCCESS;
+    if (readonly) {
+        return remountReadOnlyInContainer(result);
+    } else {
+        return ReturnCode::SUCCESS;
+    }
```
Comment (DunderRoffe) [inline libsoftwarecontainer/src/container.cpp:593]: Could skip the else clause

File: libsoftwarecontainer/src/container.cpp [hunk 3]
```diff
@@ -601,7 +605,31 @@ ReturnCode Container::bindMountFolderInContainer(const std::string &pathOnHost,
     }
 
     result = gatewaysDirInContainer() + "/" + pathInContainer;
-    return ReturnCode::SUCCESS;
+
+    if (readonly) {
+        return remountReadOnlyInContainer(result);
+    } else {
+        return ReturnCode::SUCCESS;
+    }
```
Comment (DunderRoffe) [inline libsoftwarecontainer/src/container.cpp:625]: Could skip the else clause

File: libsoftwarecontainer/src/container.cpp (hunk 1 above)
Comment (tobsan) [reply libsoftwarecontainer/src/container.cpp]: Good point, I'll remove it.

File: libsoftwarecontainer/src/container.cpp (hunk 2 above)
Comment (tobsan) [reply libsoftwarecontainer/src/container.cpp:593]: Ok

File: libsoftwarecontainer/src/container.cpp (hunk 3 above)
Comment (tobsan) [reply libsoftwarecontainer/src/container.cpp:625]: Ok

**Pull request merged**